Infrastructure: add flatpak pipelines

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Add flatpak to CI builds
#### Motivation for adding to Mudlet
Getting Flatpak up and running
#### Other info (issues closed, discussion etc)

vadi2 committed:
WIP

vadi2 committed:
Get most of dependencies in place - Mudlet can start building

vadi2 committed:
Add cleanup steps

vadi2 committed:
Comment out buggy rename-desktop

vadi2 committed:
Build completes OK

vadi2 committed:
Cleanup

vadi2 committed:
Add trailing newline

vadi2 committed:
Merge branch 'development' into add-flatpak

vadi2 committed:
update boost to 1.83.0 and enable RelWithDebInfo build

vadi2 committed:
Fix PCRE build

vadi2 committed:
Remove invalid cmake param

vadi2 committed:
Add a sample flatpak workflow

vadi2 committed:
Use another action for CI job

vadi2 committed:
Fix path to flatpak .yml

vadi2 committed:
Add desktop file into flatpak

vadi2 committed:
Try and copy mudlet.desktop into applications

Commit:
Install mudlet.desktop to share/applications

Commit:
Remove unneeded post-install call for mudlet.desktop

Commit:
Fix location to mudlet.desktop

Commit:
Enable utf8 and unicode in PCRE

Commit:
Update base image to better match our runtime

vadi2 committed:
Merge branch 'development' into add-flatpak

Commit:
Align checkout call with original action

Commit:
Add verbose key

Commit:
Remove branch key - it does not seem to be used by other flatpak-builder users

Commit:
Make pcre8 config opts be a string instead of a list

Commit:
Convert config-opts back to a list

vadi2 committed:
Use local directory as source for flatpak, not Git

vadi2 committed:
Rename app-id to id and add rename-icon

vadi2 committed:
Add icons to cmake install

vadi2 committed:
Fix rename-icon call

vadi2 committed:
Default to Wayland

vadi2 committed:
Rename mudlet.desktop to org.mudlet.mudlet.desktop

Flatpak best practices

vadi2 committed:
Revert "Rename mudlet.desktop to org.mudlet.mudlet.desktop"

This reverts commit 201f70641ce47a12b17b65bcf0c44500a937897d.

vadi2 committed:
Add alternative pipeline

vadi2 committed:
Merge branch 'development' into add-flatpak-pipelines

vadi2 committed:
Adjust triggers for second pipeline

vadi2 committed:
Build Flatpak repo on all pushes

vadi2 committed:
Remove variable interpolation in env

vadi2 committed:
Add tmate debugging

vadi2 committed:
Remove redundant `push:` trigger

vadi2 committed:
Encode private key in base64

vadi2 committed:
Remove tmate session

vadi2 committed:
Fix path to Mudlet yaml

vadi2 committed:
refactor: Use flatpak-github-actions action for flatpak build

Replaced the manual flatpak build, bundle and repo update steps with the flatpak/flatpak-github-actions/flatpak-builder@v6 action. This simplifies the workflow configuration.

Removed the GitHub pages deployment preparation commands since they are no longer needed.

vadi2 committed:
Add gpg to first flatpak workflow

vadi2 committed:
Check version of pinentry

vadi2 committed:
Install pinentry-tty

vadi2 committed:
Correct apt install

vadi2 committed:
Try with yum instead

vadi2 committed:
yum install -y

vadi2 committed:
Run yum in quiet mode

vadi2 committed:
Try installing pinentry first

ZookaOnGit commented:
What is the status on the flatpak situation?  Didn't we have some movement on this a few months ago?

vadi2 commented:
It's possible to run Mudlet as flatpak locally, but it's not available on the flathub distribution website - this PR needs some love to get it over the line.

vadi2 committed:
Merge branch 'development' into add-flatpak-pipelines

szydell commented:
Sending a bucket full of love and positive vibes! 💚 Any chance this could get some attention soon? Would love to see Mudlet on Flathub.

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-flatpak-pipelines

# Conflicts:
#	src/CMakeLists.txt

vadi2 committed:
Fix typo in gpg import flag

vadi2 commented:
Working through on this as time permits.

vadi2 committed:
Merge branch 'development' into add-flatpak-pipelines

Lazorne commented:
Feel free to contact us in the RetroDECK Team if you need any help :) 
We have the most complicated flatpak in the business. 

https://retrodeck.readthedocs.io/en/latest/blog/2025/11/25/november-2025-finally-a-fatpak/

Some our team might be able to give you advice. 

Where are you going to save the profiles and plugins and other such data where the user should manipulate it $XDG_DATA_HOME ?

Is there any reason you are targeting org.kde.Platform 6.8 and not 6.10 as your app will get flagged on Flathub.  

_Info: org.kde.Platform is end-of-life, with reason: We strongly recommend moving to the latest stable version of the Platform and SDK_

Also in other news the Mudlet integration into RetroDECK is going well ❤️

vadi2 committed:
infra: Update flatpak pipelines and dependencies

- Switch to Flathub-maintained container images (ghcr.io/flathub-infra)
- Update to KDE runtime 6.10
- Update flatpak-github-actions to v6.6
- Update actions/checkout to v4
- Use dnf instead of yum (Fedora-based containers)
- Update dependencies: libsecret 0.21.7, libzip 1.11.4, pugixml 1.15,
  luarocks 3.12.2, sqlite 3.51.2, boost 1.90.0

vadi2 committed:
add: Support XDG paths in Flatpak and Snap environments

Detect sandboxed environments via FLATPAK_ID or SNAP environment
variables and use QStandardPaths to respect their XDG path overrides.
This ensures Mudlet stores data in the proper sandboxed location
(e.g. ~/.var/app/org.mudlet.mudlet/) instead of ~/.config/mudlet.

vadi2 committed:
infra: Simplify flatpak CI and add AppStream metadata

vadi2 committed:
infra: Fix flatpak manifest for Flathub submission

vadi2 committed:
fix: Update donate URL in metainfo

vadi2 committed:
infra: Update lua-yajl to 2.1-0

vadi2 committed:
infra: Revert lua-yajl to 2.0-1 (2.1 has no release tag)

vadi2 committed:
infra: Fix libsecret meson option (gcrypt -> crypto)

vadi2 committed:
infra: Fix yajl CMake version compatibility

vadi2 committed:
infra: Fix yajl CMP0026 policy for LOCATION property

vadi2 committed:
infra: Add patch to fix yajl CMake for modern CMake

vadi2 committed:
infra: Fix yajl patch line numbers

vadi2 committed:
infra: Fix yajl patch with correct line numbers

vadi2 committed:
infra: Use tarball for libsecret instead of git

GitLab GNOME returns 403 in CI. Use tarball from download.gnome.org.

vadi2 committed:
infra: Fix Boost installation in Flatpak manifest

Install headers and patch CMake to use FindBoost module mode.

vadi2 committed:
Merge branch 'development' into add-flatpak-pipelines

Assisted-by: Claude:claude-opus-4-8

vadi2 committed:
Fix: build bundled QtKeychain in Flatpak

The Flatpak sandbox has no system Qt6Keychain, so CMake configure failed at find_package(Qt6Keychain REQUIRED). Enable USE_OWN_QTKEYCHAIN so Mudlet builds the bundled 3rdparty/qtkeychain submodule (libsecret backend is already a manifest module).

Assisted-by: Claude:claude-opus-4-8

vadi2 committed:
Fix: disable 3D mapper in Flatpak build

The 3D mapper needs assimp, which isn't in the Flatpak runtime. Mudlet's other CI builds set WITH_3DMAPPER=no, so match them and disable USE_3DMAPPER instead of bundling assimp.

Assisted-by: Claude:claude-opus-4-8

vadi2 committed:
Fix: add qtkeychain module to Flatpak, drop USE_OWN_QTKEYCHAIN

USE_OWN_QTKEYCHAIN doesn't work on Linux (its add_subdirectory is APPLE-gated), so build qtkeychain as its own Flatpak module (the sandbox equivalent of the qtkeychain-qt6-dev apt package) and let find_package(Qt6Keychain) use it.

Assisted-by: Claude:claude-opus-4-8

vadi2 committed:
Merge remote-tracking branch 'origin/development' into add-flatpak-pipelines

Assisted-by: Claude:claude-opus-4-8
